programs/KJVDataDump/main.cpp

```diff
@@ -371,14 +396,44 @@ int main(int argc, char *argv[])
 						if (bOutputTemplates) {
 							std::cout << pVerse->m_strTemplate.toUtf8().data() << std::endl;
 						}
-						if (bOutputVerseText) {
-							if (bOutputTemplates) std::cout << strSpacer.toUtf8().data();
-							if (bOutputTransChangeAdded) {
-								std::cout << lstVerseWords.join(QChar(' ')).toUtf8().data();
-							} else {
-								std::cout << strParsedVerse.toUtf8().data();
+						if (bOutputVerseText && nLetterCount > 0) {
+							QString verse = strParsedVerse.toUtf8().data();
+							QString clean = QString(verse).toLower().simplified().replace(QRegularExpression("[\\[\\]()\.,:;!?]"), "");
+							QString cleanNoSpaces = QString(clean).remove(' ');
+							int basicVowelCount = clean.count(QRegularExpression("[aeiou]"));
```

Comment:
Note that in order to support foreign language databases correctly, you'll want to make use of the word variations that KJPBS makes internally available that contains decomposed forms.

The decomposed word forms break the down individual characters into subcharacters of letter and accent mark.  Otherwise, your code here won't match vowels and such that have an accent and your counts will be skewed.

KJPBS already has code to decompose words (and internally stores them for quick reference) to handle foreign databases and do things like "accent sensitive searches".

Reply:
Excellent.  Thanks for this.  I was wondering about this.